source/common/router/upstream_request.h

```diff
@@ -107,6 +107,7 @@ class UpstreamRequest : public Logger::Loggable<Logger::Id::router>,
   UpstreamRequest* upstreamRequest() override { return this; }
 
   void clearRequestEncoder();
+  void onStreamMaxDurationReached();
```

Comment:
Please merge master to pull in @alyssawilk changes for upstream CONNECT. I suspect we need this timeout to be implemented there also but I'm not 100% sure.

Reply:
Dealt with adding CONNECT termination with max stream duration. Is this okay? cc @alyssawilk